src/Microsoft.AspNetCore.Mvc.Core/Internal/RedirectResultExecutor.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class RedirectResultExecutor
+    {
+        private readonly ILogger _logger;
+        private readonly IUrlHelperFactory _urlHelperFactory;
+
+        public RedirectResultExecutor(ILoggerFactory loggerFactory, IUrlHelperFactory urlHelperFactory)
+        {
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            if (urlHelperFactory == null)
+            {
+                throw new ArgumentNullException(nameof(urlHelperFactory));
+            }
+
+            _logger = loggerFactory.CreateLogger<RedirectResultExecutor>();
+            _urlHelperFactory = urlHelperFactory;
+        }
+
+        public void Execute(ActionContext context, RedirectResult result)
+        {
+            var urlHelper = result.UrlHelper ?? _urlHelperFactory.GetUrlHelper(context);
+
+            // IsLocalUrl is called to handle  Urls starting with '~/'.
```

Comment:
Like your suggestion @pranavkm thought it's actually "IsLocalUrl and Content are called to resolve local URLs to absolute URLs.". In particular, it's not just `IsLocalUrl()`.